# Guided Project 3: Exploring Ebay Car Sales Data (Numpy and Pandas)

Many eBay users use the website to buy and sell used cars. [This][1] source on Kaggle provides detailed data of such listings from *eBay Kleinanzeigen* - a classified section of the German eBay website.

The purpose of this project is to clean this data and perform analysis using newly acquired knowledge of the numpy and pandas libraries.


[1]: https://www.kaggle.com/orgesleka/used-cars-database/data

## Step 1: Import data and preliminary exploration

The following cell imports `numpy` and `pandas` libraries, before importing the .csv data as a pandas dataframe and saving in variable `autos`. The basic information on the dataframe and the firsts 5 entried are then printed to make first observations on data.

In [1]:
import numpy as np
import pandas as pd
from pprint import *
autos = pd.read_csv("autos.csv", encoding = "Latin-1")

print(autos.info())
print(autos.head(5))

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 50000 entries, 0 to 49999
Data columns (total 20 columns):
 #   Column               Non-Null Count  Dtype 
---  ------               --------------  ----- 
 0   dateCrawled          50000 non-null  object
 1   name                 50000 non-null  object
 2   seller               50000 non-null  object
 3   offerType            50000 non-null  object
 4   price                50000 non-null  object
 5   abtest               50000 non-null  object
 6   vehicleType          44905 non-null  object
 7   yearOfRegistration   50000 non-null  int64 
 8   gearbox              47320 non-null  object
 9   powerPS              50000 non-null  int64 
 10  model                47242 non-null  object
 11  odometer             50000 non-null  object
 12  monthOfRegistration  50000 non-null  int64 
 13  fuelType             45518 non-null  object
 14  brand                50000 non-null  object
 15  notRepairedDamage    40171 non-null  object
 16  date

The following observations are made from this initial exploration of the data:
- There are 20 columns to the dataframe
- There are 50,000 entries, although 5 columns contain null data:
    1. `vehicleType`
    2. `gearbox`
    3. `model`
    4. `fuelType`
    5. `notRepairedDamage`
    
- The data in series exists as either `object` (i.e. data is a string) or `int64` (i.e. data is integer numbers) data types
- Data that descibes a date/time is stored as a string
- data stored in columns `price` and `odometer` are of `object` data type, and should be cleaned to numeric data type to allow analysis
- The text is in German
- Column names are in camelcase, rather than snakecase

## Step 2: Convert column names from camelcase to snakecase




The incumbent column names are saved to a variable `camel_cols`.

A function - `change_case` - is created takes a string as an input (should be in camelcase) and returns the string in snakecase.

A for loop is created which corrects the following specific entries (if they exist):
- `"yearOfRegistration"` to `"registration_year"`
- `"monthOfRegistration"` to `"registration_month"`
- `"notRepairedDamage"` to `"unrepaired_damage"`
- `"dateCreated"` to `"ad_created"`

If the incumbent column header is not one of the above instances, then `change_case` is applied to ensure all column headers are in snakecase. The corrected headers are printed along with the first 3 entries.




In [2]:
camel_cols = autos.columns.astype(str)
print ("""\033[1m Columns in Camel Case: \033[0m
""", camel_cols)

def change_case(str): 
    res = [str[0].lower()] 
    for c in str[1:]: 
        if c in ('ABCDEFGHIJKLMNOPQRSTUVWXYZ'): 
            res.append('_') 
            res.append(c.lower()) 
        else: 
            res.append(c) 
      
    return ''.join(res) 

snake_cols = []
for col in camel_cols:
    if col == "yearOfRegistration":
        snake_col = "registration_year"
    elif col == "monthOfRegistration":
        snake_col = "registration_month"
    elif col == "notRepairedDamage":
        snake_col = "unrepaired_damage"
    elif col == "dateCreated":
        snake_col = "ad_created"
    else:
        snake_col = change_case(col)
    snake_cols.append(snake_col)

print("""
\033[1m Corrected Columns in Snake Case: \033[0m
""")#, snake_cols)

autos.columns = snake_cols
print(autos.head(3))

 Columns in Camel Case: 
 Index(['dateCrawled', 'name', 'seller', 'offerType', 'price', 'abtest',
       'vehicleType', 'yearOfRegistration', 'gearbox', 'powerPS', 'model',
       'odometer', 'monthOfRegistration', 'fuelType', 'brand',
       'notRepairedDamage', 'dateCreated', 'nrOfPictures', 'postalCode',
       'lastSeen'],
      dtype='object')

 Corrected Columns in Snake Case: 

          date_crawled                                        name  seller  \
0  2016-03-26 17:47:46            Peugeot_807_160_NAVTECH_ON_BOARD  privat   
1  2016-04-04 13:38:56  BMW_740i_4_4_Liter_HAMANN_UMBAU_Mega_Optik  privat   
2  2016-03-26 18:57:24                  Volkswagen_Golf_1.6_United  privat   

  offer_type   price   abtest vehicle_type  registration_year    gearbox  \
0    Angebot  $5,000  control          bus               2004    manuell   
1    Angebot  $8,500  control    limousine               1997  automatik   
2    Angebot  $8,990     test    limousine               2009    manuel

## Step 3: Detailed exploration

Pandas `describe` method is used to give a more detailed insight to the dataframe contents.

In [3]:
autos.describe(include = "all")

date_crawled         name  seller offer_type  price abtest  \
count                 50000        50000   50000      50000  50000  50000   
unique                48213        38754       2          2   2357      2   
top     2016-04-04 16:40:33  Ford_Fiesta  privat    Angebot     $0   test   
freq                      3           78   49999      49999   1421  25756   
mean                    NaN          NaN     NaN        NaN    NaN    NaN   
std                     NaN          NaN     NaN        NaN    NaN    NaN   
min                     NaN          NaN     NaN        NaN    NaN    NaN   
25%                     NaN          NaN     NaN        NaN    NaN    NaN   
50%                     NaN          NaN     NaN        NaN    NaN    NaN   
75%                     NaN          NaN     NaN        NaN    NaN    NaN   
max                     NaN          NaN     NaN        NaN    NaN    NaN   

       vehicle_type  registration_year  gearbox     power_p_s  model  \
count         44905       50000.000000    47320  50000.000000  47242   
unique            8                NaN        2           NaN    245   
top       limousine                NaN  manuell           NaN   golf   
freq          12859                NaN    36993           NaN   4024   
mean            NaN        2005.073280      NaN    116.355920    NaN   
std             NaN         105.712813      NaN    209.216627    NaN   
min             NaN        1000.000000      NaN      0.000000    NaN   
25%             NaN        1999.000000      NaN     70.000000    NaN   
50%             NaN        2003.000000      NaN    105.000000    NaN   
75%             NaN        2008.000000      NaN    150.000000    NaN   
max             NaN        9999.000000      NaN  17700.000000    NaN   

         odometer  registration_month fuel_type       brand unrepaired_damage  \
count       50000        50000.000000     45518       50000             40171   
unique         13                 NaN         7          40                 2   
top     150,000km                 NaN    benzin  volkswagen              nein   
freq        32424                 NaN     30107       10687             35232   
mean          NaN            5.723360       NaN         NaN               NaN   
std           NaN            3.711984       NaN         NaN               NaN   
min           NaN            0.000000       NaN         NaN               NaN   
25%           NaN            3.000000       NaN         NaN               NaN   
50%           NaN            6.000000       NaN         NaN               NaN   
75%           NaN            9.000000       NaN         NaN               NaN   
max           NaN           12.000000       NaN         NaN               NaN   

                 ad_created  nr_of_pictures   postal_code            last_seen  
count                 50000         50000.0  50000.000000                50000  
unique                   76             NaN           NaN                39481  
top     2016-04-03 00:00:00             NaN           NaN  2016-04-07 06:17:27  
freq                   1946             NaN           NaN                    8  
mean                    NaN             0.0  50813.627300                  NaN  
std                     NaN             0.0  25779.747957                  NaN  
min                     NaN             0.0   1067.000000                  NaN  
25%                     NaN             0.0  30451.000000                  NaN  
50%                     NaN             0.0  49577.000000                  NaN  
75%                     NaN             0.0  71540.000000                  NaN  
max                     NaN             0.0  99998.000000                  NaN

We can see from this information that some columns contain almost exclusively one value, and so should be considered to be dropped. Those are:
- `seller`
- `offer_type`

The contents of some columns is unclear, and should be investigated further to identify their potential usefulness:
- `nr_of_pictures` : unclear what use this information gives

The following series should be cleaned so that the data may be stored numberically, rather than as a string:
- `price`
- `odometer`

In [4]:
print("\033[1m'seller' column values count:\033[0m")
print(autos["seller"].value_counts())

print("\033[1m'offer_type' column values count:\033[0m")
print(autos["offer_type"].value_counts())

'seller' column values count:
privat        49999
gewerblich        1
Name: seller, dtype: int64
'offer_type' column values count:
Angebot    49999
Gesuch         1
Name: offer_type, dtype: int64


There is only a single entry for listings with a "gewerblich" type seller (translates as "commercial").
Similarly, only one offer type is listed as "Gesuch" (translates as "request"). 
Theses entries should each be deleted, as they offer no use for analysis.

In [5]:
autos = autos.loc[autos["seller"] != "gewerblich", :]
autos = autos.loc[autos["offer_type"] != "Gesuch", :]

Now looking closer into the entries under the column `nr_of_pictures`:

In [6]:
print(autos["nr_of_pictures"].value_counts())

0    49998
Name: nr_of_pictures, dtype: int64


This column is clearly redundant as all values are equal to 0. Therefore, this column should be dropped.

In [7]:
autos = autos.drop(labels = "nr_of_pictures", axis=1)

Finally, the data in columns `price` and `odometer` should be changed to a numeric type to facilitate analysis.

In [8]:
autos["price"] = autos["price"].str.replace("$", "")
autos["price"] = autos["price"].str.replace(",", "")
autos["price"] = autos["price"].astype(int)

autos["odometer"] = autos["odometer"].str.replace("km", "")
autos["odometer"] = autos["odometer"].str.replace(",", "")
autos["odometer"] = autos["odometer"].astype(int)
autos.rename({"odometer": "odometer_km"}, axis = 1, inplace = True)

## Step 4: Clean `price` and `odometer_km` data
To ensure the data for `price` and `odometer_km` is high quality, further exploration is performed below to give greater transparency into these data sets.

In [9]:
print("MIN PRICE :", autos["price"].min())
print("MAX PRICE :", autos["price"].max())
print("MIN ODOMETER (km) :", autos["odometer_km"].min())
print("MAX ODOMETER (km) :", autos["odometer_km"].max())
print("---")
print ("PRICE UNIQUE VALS :", autos["price"].unique().shape)
print ("ODOMETER UNIQUE VALS :", autos["odometer_km"].unique().shape)
print("---")
print("\033[1mPRICE DESC :\033[0m")
print(autos["price"].describe())
print("---")
print("\033[1mODOMETER DESC :\033[0m")
print(autos["odometer_km"].describe())
print("---")
print("\033[1mPRICE FT (ASCENDING) :\033[0m")
print(autos["price"].value_counts().sort_index(ascending = True).head(100))

print("\033[1mPRICE FT (DESCENDING) :\033[0m")
print(autos["price"].value_counts().sort_index(ascending = False).head(15))
print("---")
print("\033[1mODOMETER FT :\033[0m")
print(autos["odometer_km"].value_counts())

MIN PRICE : 0
MAX PRICE : 99999999
MIN ODOMETER (km) : 5000
MAX ODOMETER (km) : 150000
---
PRICE UNIQUE VALS : (2357,)
ODOMETER UNIQUE VALS : (13,)
---
PRICE DESC :
count    4.999800e+04
mean     9.840435e+03
std      4.811140e+05
min      0.000000e+00
25%      1.100000e+03
50%      2.950000e+03
75%      7.200000e+03
max      1.000000e+08
Name: price, dtype: float64
---
ODOMETER DESC :
count     49998.000000
mean     125731.729269
std       40042.718425
min        5000.000000
25%      125000.000000
50%      150000.000000
75%      150000.000000
max      150000.000000
Name: odometer_km, dtype: float64
---
PRICE FT (ASCENDING) :
0      1420
1       156
2         3
3         1
5         2
       ... 
300     384
310       1
320      12
325       5
329       2
Name: price, Length: 100, dtype: int64
PRICE FT (DESCENDING) :
99999999    1
27322222    1
12345678    3
11111111    2
10000000    1
3890000     1
1300000     1
1234566     1
999999      2
999990      1
350000      1
345000      1
299

This information allows us to see that there are many outliers in the `price` data. There are vehicles listed at very low prices (minimum being \\$0) and at very high prices (maximum being \\$99,999,999).

It was chosen to treat all vehicles costing less than \\$250 or costing more than \\$500,000 as outliers, and hence they are removed from the `autos` data set below.

In [10]:
autos = autos.loc[autos["price"].between(250, 500000), :]

print("\033[1mPRICE DESC :\033[0m")
print(autos["price"].describe())

PRICE DESC :
count     47288.000000
mean       6044.447217
std        9131.139284
min         250.000000
25%        1300.000000
50%        3200.000000
75%        7551.250000
max      350000.000000
Name: price, dtype: float64


Having removed these outliers, we can see that the mean price and is significantly lower, and that the standard deviation of the remaining data is also lower. 
There are still ample data points (47,288) to be able to perform significant analysis.

## Step 5: Examine date-based data

A brief overview of the dates contained in columns `date_crawled`, `ad_created` and `last_seen` is given here by truncating the first ten characters of the strings contained in these data points (this gives the date in long format i.e. yyyy-mm-dd).
The distribution of these values can then be evaluated.

In [11]:
date_crawled_dates = autos["date_crawled"].str[:10]
print(date_crawled_dates.value_counts(normalize = True, dropna = False).sort_index())

ad_created_dates = autos["ad_created"].str[:10]
print(ad_created_dates.unique().shape)
print(ad_created_dates.value_counts(normalize = True, dropna = False).sort_index())

last_seen_dates = autos["last_seen"].str[:10]
print(last_seen_dates.unique().shape)
print(last_seen_dates.value_counts(normalize = True, dropna = False).sort_index())

2016-03-05    0.025313
2016-03-06    0.014084
2016-03-07    0.036183
2016-03-08    0.033095
2016-03-09    0.032947
2016-03-10    0.032461
2016-03-11    0.032778
2016-03-12    0.036986
2016-03-13    0.015628
2016-03-14    0.036457
2016-03-15    0.034174
2016-03-16    0.029458
2016-03-17    0.031488
2016-03-18    0.012836
2016-03-19    0.034618
2016-03-20    0.037980
2016-03-21    0.037430
2016-03-22    0.032778
2016-03-23    0.032270
2016-03-24    0.029246
2016-03-25    0.031298
2016-03-26    0.032291
2016-03-27    0.031107
2016-03-28    0.035083
2016-03-29    0.033772
2016-03-30    0.033835
2016-03-31    0.031847
2016-04-01    0.033856
2016-04-02    0.035675
2016-04-03    0.038805
2016-04-04    0.036563
2016-04-05    0.013132
2016-04-06    0.003172
2016-04-07    0.001353
Name: date_crawled, dtype: float64
(76,)
2015-06-11    0.000021
2015-08-10    0.000021
2015-09-09    0.000021
2015-11-10    0.000021
2015-12-05    0.000021
                ...   
2016-04-03    0.039059
2016-04-04    0.

This exploration shows us that the data in `date_crawled` and `last_seen` is taken on a series of days between 2016-03-05 and 2016-04-07. The `date_crawled` data is fairly evenly spread across these days (approximately 3% of entries per day). The `last_seen` data is more skewed towards the final dates of in this period (approximately 50% of dates are within the last 3 dates of this period). There are 34 unique dates for both these sets of data.

The `ad_created` data stretches over a much broader range - from 2015-06-11 to 2016-04-07. Clearly, this shows that some vehicles were listed on the website a long time before the data was started to be crawled. However, the much older entries account for only very few of the data points, with a much higher percentage of listings being created during (and likely just before) the data was crawled. There are a much higher number of dates listed in this column (76).

Below a brief insight to the data listed in `registration_year` is given.

In [12]:
autos["registration_year"].describe()

count    47288.000000
mean      2004.844971
std         88.753302
min       1000.000000
25%       1999.000000
50%       2004.000000
75%       2008.000000
max       9999.000000
Name: registration_year, dtype: float64

Clearly no vehicles were registered in the year 1000 or in the year 9999. Therefore, some more cleaning is required to remove outlying data from this column. Below an overview of the oldest cars in the dataset, all registered before 1950 is given.

In [13]:
before_1950 = autos.loc[autos["registration_year"] < 1950, :]
before_1950 =before_1950.sort_values("registration_year", ascending = False)
print (before_1950)

              date_crawled                                               name  \
11047  2016-03-08 20:50:10          Andere_Simca_5_Fourgonette_Kombilimousine   
11585  2016-03-11 21:48:36                              Volkswagen__VW_Typ_82   
13963  2016-03-20 17:51:49  Mercedes_Benz_L1500S_Wehrmacht_/_Luftwaffe___F...   
25792  2016-03-09 10:54:11              Andere_GAZ_Oldtimergelaendewagen_1941   
24855  2016-03-31 13:57:18                      Andere_DKW_F8_Cabrio_Limosine   
26103  2016-03-16 12:37:38               Oldtimer_Sport___Cabrio_Georges_Irat   
39725  2016-03-25 22:54:31  Dodge_1937_Hotrod_Flathead_Scheunenfund_Oldtim...   
21421  2016-03-05 17:45:32            Ford_Business_Coupe_Hotrod_Projekt.1937   
23804  2016-03-30 22:42:50                                 Mercedes_Benz_170V   
26607  2016-03-21 14:45:26                                          MG_Andere   
2221   2016-03-15 14:57:07  Sehr_seltener_Oldtimer_Opel_1210_zum_Restaurieren   
2573   2016-03-19 22:51:25  

A brief investigation using the `name` column to identify if the cars listed really were registered in the date suggested shows that all cars up to and including the "Essex_super_six__Ford_A" registered in 1927 are likely to have been listed in the year listed under `registration_date`.

Therefore, the low-end cutoff year to be used is 1925. No cars listed should be registered in future dates, so the years 2017 onwards will also be treated as outliers.

In [14]:
autos = autos.loc[autos["registration_year"].between (1925, 2016), :]
autos["registration_year"].describe()

count    45436.000000
mean      2003.032529
std          7.078276
min       1927.000000
25%       1999.000000
50%       2003.000000
75%       2008.000000
max       2016.000000
Name: registration_year, dtype: float64

## Step 6: Aggregation analysis by brand

First we need to determine which car manufacturers are the most appropriate to focus this analysis on. Let's start by reviewing the frequency of entries by manufacturer.

In [15]:
brand_val_counts = autos["brand"].value_counts(normalize= True)
print(brand_val_counts)

volkswagen        0.211330
bmw               0.111850
opel              0.104895
mercedes_benz     0.098314
audi              0.088146
ford              0.068008
renault           0.046065
peugeot           0.029910
fiat              0.024958
seat              0.018201
skoda             0.016617
nissan            0.015208
mazda             0.015164
smart             0.014482
citroen           0.014130
toyota            0.013029
hyundai           0.010146
sonstige_autos    0.009574
volvo             0.009266
mini              0.008936
mitsubishi        0.008165
honda             0.007945
kia               0.007197
alfa_romeo        0.006669
porsche           0.006141
suzuki            0.005832
chevrolet         0.005766
chrysler          0.003565
dacia             0.002707
daihatsu          0.002377
jeep              0.002333
land_rover        0.002157
subaru            0.002069
saab              0.001651
jaguar            0.001541
daewoo            0.001387
rover             0.001343
t

Although it is clear from this brief overview of the data that a high percentage of entries belong to just a small handful of manufacturers, to provide interesting analysis all manufacturers responsible for >1% of the entries will be used in this analysis.

In [37]:
agg_brands = brand_val_counts[brand_val_counts > 0.01].index

brand_price_dict = {}
for brand in agg_brands:
    brand_price_dict[brand] = autos.loc[autos["brand"] == brand, "price"].mean()
brand_price_sorted = sorted(brand_price_dict.items(), key=lambda x:x[1], reverse=True)
for brand in brand_price_sorted:
    print("{0}: ${1:,.0f}".format(brand[0], brand[1]))

audi: $9,420
mercedes_benz: $8,697
bmw: $8,422
skoda: $6,459
volkswagen: $5,545
hyundai: $5,445
toyota: $5,176
nissan: $4,889
seat: $4,531
mazda: $4,228
ford: $3,951
citroen: $3,847
smart: $3,596
peugeot: $3,168
opel: $3,127
fiat: $2,962
renault: $2,595


The aggrategate analysis of mean brand prices shown above shows that there is significant variation in mean brand cost across the dataset. The most expensive brand of car, Audi, costs on average over 3.5X more than the least expensive brand, Renault.

This insight would be particularly helpful for highly price-conscious customers, seeking to buy cheaper car brands. It would enable them to refine theor search to focus on car brands that are likely to be in their price range.

## Step 7: Aggregate analysis by mileage

The same analysis as above is performed on the same most common brands to give information pertaining to the mean mileage by brand.

In [38]:
brand_odo_dict = {}
for brand in agg_brands:
    brand_odo_dict[brand] = autos.loc[autos["brand"] == brand, "odometer_km"].mean()
brand_odo_sorted = sorted(brand_odo_dict.items(), key=lambda x:x[1], reverse=True)
for brand in brand_odo_sorted:
    print("{0}: {1:,.0f} km".format(brand[0], brand[1]))

bmw: 132,752 km
mercedes_benz: 131,144 km
audi: 129,252 km
opel: 129,036 km
volkswagen: 128,687 km
renault: 127,735 km
peugeot: 126,726 km
mazda: 124,122 km
ford: 123,995 km
seat: 121,608 km
citroen: 119,657 km
nissan: 118,003 km
fiat: 116,772 km
toyota: 116,106 km
skoda: 111,040 km
hyundai: 106,605 km
smart: 99,734 km


It is challenging to combine both insights on brand prices and brand mileages to give a more informative analysis of the brand pricing. Therefore, further processing of this data is necessary.

## Step 8: Combining price and mileage data into a single dataset

Below the aggragated data on pricing is stored in a new series `brand_price_series`, and the aggragated data on milage is stored in the series `brand_odo_series`.

These series are then combined into a single dataframe `price_odo_df`, with the brand names acting as the index.

The dataframe is printed with styling applied to facilitate visual analysis.

In [39]:
brand_price_series = pd.Series(brand_price_dict)
brand_odo_series = pd.Series(brand_odo_dict)

price_odo_df = pd.DataFrame(brand_price_series, columns = ["price"])

price_odo_df["odometer_km"] = brand_odo_series

price_odo_df = price_odo_df.sort_values("price", ascending = False)

price_odo_df

import seaborn as sns

cm = sns.light_palette("green", as_cmap=True)

s = price_odo_df.style.background_gradient(cmap=cm)
s

The printed dataframe above is much more visually aesthetic and well suited to visual analysis.

We can see that the most expensive manufacturers, `audi`, `mercedes_benz` and `bmw` on average have the highest mileage, making them particularly bad value for money in this aspect.

The least expensive manufacturers `peugeot`, `opel`, `fiat` and `renault` also have high mean mileages.

In terms of value for money when considering milage, the brand that would be most recommended from this analysis is `smart`- This is the 5th least expensive manufacturer is the only brand that has, on average, less than 100,000 km.

## Step 9: Further Analysis

Below the most common model for each brand is calculated, along with the percentage of listings for that brand accounted for by the most common model.

In [95]:
tot_brands = brand_val_counts.index

for brand in tot_brands:
    brand_pop_mod = autos.loc[autos["brand"] == brand, "model"].mode(dropna=False)
    brand_series = autos.loc[autos["brand"] == brand, :]
    perc_pop_model = 100*(brand_series.loc[autos["model"] == brand_pop_mod[0] , "model"].count()/brand_series["model"].count())
    print (brand, ":", brand_pop_mod[0], "{:,.2f}%".format(perc_pop_model))

volkswagen : golf 39.05%
bmw : 3er 53.04%
opel : corsa 31.81%
mercedes_benz : c_klasse 26.07%
audi : a4 31.33%
ford : focus 25.20%
renault : twingo 28.08%
peugeot : 2_reihe 46.75%
fiat : punto 35.38%
seat : ibiza 39.41%
skoda : octavia 40.95%
nissan : micra 34.13%
mazda : 6_reihe 24.46%
smart : fortwo 86.69%
citroen : andere 34.09%
toyota : yaris 21.88%
hyundai : i_reihe 40.09%
sonstige_autos : nan nan%
volvo : v70 21.76%


C:\Users\tomel\anaconda3\lib\site-packages\ipykernel_launcher.py:6: RuntimeWarning: invalid value encountered in long_scalars
  


mini : cooper 66.17%
mitsubishi : colt 29.83%
honda : civic 48.22%
kia : andere 20.58%
alfa_romeo : 156 28.97%
porsche : 911 50.00%
suzuki : andere 44.36%
chevrolet : andere 58.63%
chrysler : andere 33.77%
dacia : sandero 36.36%
daihatsu : cuore 54.29%
jeep : grand 43.00%
land_rover : freelander 31.96%
subaru : legacy 31.82%
saab : andere 73.61%
jaguar : andere 55.07%
daewoo : matiz 33.90%
rover : andere 93.10%
trabant : 601 84.31%
lancia : ypsilon 40.00%
lada : niva 60.00%


This breakdown shows that the most common `smart` model is the `smart fortwo`, responsible for 87% of the listings for this brand. Therefore, for buyers looking to get a cheap Smart car with low mileage, the most widely available is likely to be the `smart fortwo`.